qml/Ping360Visualizer.qml

```diff
@@ -0,0 +1,120 @@
+        chart.draw(points, length + initialPoint, initialPoint)
+    }
+
+    function setDepth(depth) {
+        readout.value = depth
+    }
+
+    function setConfidence(perc) {
+        readout.confidence = perc
+    }
+
+    QC1.SplitView {
+        orientation: Qt.Horizontal
+        anchors.fill: parent
+
+        Item {
+            Layout.fillHeight: true
+            Layout.fillWidth: true
+
+            PolarPlot {
+                id: waterfall
+                height: Math.min(parent.height, parent.width)
+                width: height
+                anchors.centerIn: parent
+
+                Shape {
+                    visible: waterfall.containsMouse
+                    anchors.centerIn: parent
+                    opacity: 0.5
+                    ShapePath {
+                        strokeWidth: 3
+                        strokeColor: StyleManager.secondaryColor
+                        startX: 0
+                        startY: 0
+                        //TODO: This need to be updated in sensor integration
+                        PathLine {
+                            property real angle: -Math.atan2(waterfall.mousePos.x - waterfall.width/2, waterfall.mousePos.y - waterfall.height/2) + Math.PI/2
+                            x: waterfall.width*Math.cos(angle)/2
+                            y: waterfall.height*Math.sin(angle)/2
+                        }
+                    }
+                }
+
+                Text {
+                    id: mouseReadout
+                    visible: waterfall.containsMouse
+                    x: waterfall.mousePos.x - width/2
+                    y: waterfall.mousePos.y - height*2
+                    text: (waterfall.mouseSampleDepth*SettingsManager.distanceUnits['distanceScalar']).toFixed(2) + SettingsManager.distanceUnits['distance']
+                    color: confidenceToColor(waterfall.mouseSampleConfidence)
+                    font.family: "Arial"
+                    font.pointSize: 15
+                    font.bold: true
+
+                    Text {
+                        id: mouseConfidenceText
+                        x: mouseReadout.width - width
+                        y: mouseReadout.height*4/5
+                        text: transformValue(waterfall.mouseSampleConfidence) + "%"
+                        visible: typeof(waterfall.mouseSampleConfidence) == "number"
+                        color: confidenceToColor(waterfall.mouseSampleConfidence)
+                        font.family: "Arial"
+                        font.pointSize: 10
+                        font.bold: true
+                    }
+                }
+            }
+        }
+
+        Chart {
+            id: chart
+            Layout.fillHeight: true
+            Layout.maximumWidth: 250
+            Layout.preferredWidth: 100
+            Layout.minimumWidth: 75
+        }
+
+        Settings {
+            property alias chartWidth: chart.width
+        }
+    }
+
+    ValueReadout {
```

Comment:
This ought to be removed, right? We don't have any 'depth' or 'confidence'.

Reply:
I believe that we can use it for tests and remove before merging with master.